contracts/lib/Poseidon.sol

```diff
@@ -1,21 +1,28 @@
-pragma solidity ^0.6.0;
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
 
-contract PoseidonUnit {
-  function poseidon(uint256[] memory) public pure returns(uint256) {}
```

Comment:
how many inputs this function did support before?

Reply:
In a very old version of Poseidon EVM code generator it supported arbitrary number of inputs. In the up-to-date version it supports a constant number of inputs as a generator parameter.
Dunno the reason why this was done. May be this is more efficient in terms of the gas cost.

Reply:
But we will need to have a Poseidon contract to hash an arbitrary number of inputs. what we will do in this  case?